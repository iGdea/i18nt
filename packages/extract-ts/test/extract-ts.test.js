const { describe, it, expect } = require('@jest/globals');
const { extractTs } = require('../');

describe('#extractTs', () => {
  it('#base', () => {
    const fileName = '/root/example.ts';
    const fileContent = `
    i18n('i18n text');
    i18n('i18n text');
    i18n('');
    i18n(\`\`);
    i18n('i18n text', 'subkey1');
    i18n('i18n text', 'subkey1');
    i18n('i18n text', { language: 'en', subkey: 'subkey2' });
    i18n(\`i18n text\`, [], { language: 'en', subkey: \`subkey3\` });
    i18n.jsEncode('jsEncode text', [i18n('sub i18n')]);

    i18n.t\`i18n.t msg1\`
    i18n.t\`i18n.t msg2 \${username}@\${corpname}\`
    i18n.t({ subkey: 'subkey4' })\`i18n.t msg3\`
    `;

    const result = extractTs({
      fileName,
      fileContent,
    });
    expect(result).toMatchSnapshot();
  });
});
