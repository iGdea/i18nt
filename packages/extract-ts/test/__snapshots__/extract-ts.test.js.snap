// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#extractTs #base 1`] = `
{
  "list": [
    {
      "subkey": undefined,
      "text": "i18n text",
    },
    {
      "subkey": "subkey1",
      "text": "i18n text",
    },
    {
      "subkey": "subkey2",
      "text": "i18n text",
    },
    {
      "subkey": "subkey3",
      "text": "i18n text",
    },
    {
      "subkey": undefined,
      "text": "jsEncode text",
    },
    {
      "subkey": undefined,
      "text": "sub i18n",
    },
    {
      "text": "i18n.t msg1",
    },
    {
      "text": "i18n.t msg2 %{0}@%{1}",
    },
    {
      "subkey": "subkey4",
      "text": "i18n.t msg3",
    },
  ],
}
`;
